### PR TITLE
Search bar functionality 

### LINKED ISSUE #39: Search bar functionality
**Is your feature request related to a problem? Please describe.**
We want to reset the page count when a new search happens. We want to focus on the input box when we click the search icon initially. We want to be able to submit the form when we click the search icon when the search bar is open. 

**Describe the solution you'd like**
We will reset the page count state on the form submit. Put a button on the icon to submit the form. Write a function to focus the input.

### PR DESCRIPTION
**Pull request by krferrell**
## Changes
1. Updated search bar to search on click and on enter key
2. reset page count when searching
3. autofocus input on icon click 

## Purpose
The changes outlined above will improve the user experience. 

## Approach
Added a state variable to keep track of the search string. Switched to a controlled component to handle the search features. Created functions to handle the click and keypress. Used the useRef hook to autofocus. Updated pageSet to setPage and called it on the submission of the input.   

Closes #39

### CODE REVIEW

**Review by gonzalespaulb — APPROVED** [review]
Input logic looks neat

**Review by Gjhnsn — APPROVED** [review]
All good!

File: movie-app/src/header/NavBar.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
-import { useState } from "react";
+import { useState, useRef } from "react";
```
Comment (Lap343) [inline movie-app/src/header/NavBar.js:1]: niceeeeeee

**Review by Lap343 — APPROVED** [review]
lol only place I've seen someone useRef. Good code

**Pull request merged**